url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/execution/AgeConverter.java

```diff
@@ -0,0 +1,585 @@
+        if (age < 30) {
+            return 25;
+        }
+        if (age < 45) {
+            return 30;
+        }
+        if (age < 60) {
+            return 45;
+        }
+        if (age < 65) {
+            return 60;
+        }
+        if (age < 70) {
+            return 65;
+        }
+        if (age < 75) {
+            return 70;
+        }
+        if (age < 96) {
+            return (short) age;
+        }
+        return 97;
+    }
+
+    /**
+     * @param age
+     * @param aRandom
+     * @return int age which is by default 90 else a random assignment in ranges:
+     * 0-19,20-24,25-29,30-39,40-44,45-49,50-54,55-59,60-64,65-74,75-84,85-90.
+     */
+    public static int getAge1(
+            int age,
+            Random aRandom) {
+        if (age < 20) {
+            return aRandom.nextInt(20);
+        }
+        if (age < 25) {
+            return 20 + aRandom.nextInt(5);
+        }
+        if (age < 30) {
+            return 25 + aRandom.nextInt(5);
+        }
+        if (age < 35) {
+            return 30 + aRandom.nextInt(5);
+        }
+        if (age < 40) {
+            return 35 + aRandom.nextInt(5);
+        }
+        if (age < 45) {
+            return 40 + aRandom.nextInt(5);
+        }
+        if (age < 50) {
+            return 45 + aRandom.nextInt(5);
+        }
+        if (age < 55) {
+            return 50 + aRandom.nextInt(5);
+        }
+        if (age < 60) {
+            return 55 + aRandom.nextInt(5);
+        }
+        if (age < 65) {
+            return 60 + aRandom.nextInt(5);
+        }
+        if (age < 75) {
+            return 65 + aRandom.nextInt(10);
+        }
+        if (age < 85) {
+            return 75 + aRandom.nextInt(10);
+        }
+        if (age < 90) {
+            return 85 + aRandom.nextInt(5);
+        }
+        return 90;
+    }
+
+    /**
+     * @return int age which is by default 80 else a random assignment in ranges:
+     * 0-19,20-29,30-39,40-49,50-59,60-69,70-79
+     * @param age0
+     * @param aRandom
+     */
+    public static int getAge2(
+            int age0,
+            Random aRandom) {
+        if (age0 < 20) {
+            return aRandom.nextInt(20);
+        }
+        if (age0 < 30) {
+            return (20 + aRandom.nextInt(10));
+        }
+        if (age0 < 40) {
+            return (30 + aRandom.nextInt(10));
+        }
+        if (age0 < 50) {
+            return (40 + aRandom.nextInt(10));
+        }
+        if (age0 < 60) {
+            return (50 + aRandom.nextInt(10));
+        }
+        if (age0 < 70) {
+            return (60 + aRandom.nextInt(10));
+        }
+        if (age0 < 80) {
+            return (70 + aRandom.nextInt(10));
+        }
+        return 80;
+    }
+
+    /**
+     * @param age
+     * @param tRandom
+     * @return int age which is by default 90 else a random assignment in ranges:
+     * 0-4,5-7,8-9,10-14,15,16-19,20-24,25-29,30-44,45-59,60-64,65-74,75-84,85-90.
+     */
+    public static int getAge3(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.